src/compiler/marks.ts

```diff
@@ -37,13 +33,35 @@ marks.def = function(encoding, layout, style) {
   return defs;
 };
 
-marks.bar = {
+function switchMark(mark: string): any {
+  // TODO: use constants
+  switch (mark) {
+    case 'bar':
+      return bar;
```

Comment:
should this just return 

```
{
   type: 'rect'
   prop: bar_props,
   supportedEncoding: {row: 1, col: 1, x: 1, y: 1, size: 1, color: 1}		        
 };		 
```

rather than having 
```
export var bar = ...
```

below?